### PR TITLE
fix #1629 : migrated MakeTransferFragment to compose

### LINKED ISSUE #1629: refactor: migrate MakeTransferFragment to compose
migrate MakeTransferFragment to compose

### PR DESCRIPTION
**Pull request by AdityaKumdale**
## Issue Fix
Fixes #1629 

## Screenshots
before migration:
image

after migration: 
image

##

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+        mutableStateOf(showBottomSheet)
+    }
+
+    if (showBottomSheet) {
+        ModalBottomSheet(
+            sheetState = sheetState,
+            onDismissRequest = {
+                showBottomSheet = false
+            },
+            dragHandle = { BottomSheetDefaults.DragHandle() },
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt:132]: i think its the default value ?

File: mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+    externalId: String,
+    transferAmount: Double,
+    makeTransfer: (Long, Double) -> Unit,
+    showBottomSheet: Boolean,
+    sheetState: SheetState?
+) {
+    val scope = rememberCoroutineScope()
+    var showBottomSheet by rememberSaveable {
+        mutableStateOf(showBottomSheet)
+    }
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt:165]: point of adding this here

File: mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+                                showBottomSheet = false
+                            }
+                        }
+                    }
+                },
+                modifier = Modifier
+                    .width(100.dp)
+                    .height(50.dp)
+            ) {
+                Text(text = "Cancel")
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt:266]: extract string res

File: mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+                    makeTransfer(
+                        toClientId,
+                        transferAmount
+                    )
+                },
+                modifier = Modifier
+                    .width(100.dp)
+                    .height(50.dp)
+            ) {
+                Text(text = "Confirm")
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt:282]: string res

File: mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+        ) {
+            Column(
+                modifier = Modifier
+                    .padding(20.dp)
+            ) {
+                Text(
+                    text = stringResource(id = R.string.sending_to) + Constants.COLON,
+                    style = TextStyle(
+                        Color.Black,
+                        MaterialTheme.typography.bodyMedium.fontSize
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt:198]: repetitive styles can be moved to `MifosTextStyle.kt`

File: mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt [hunk 6]
```diff
@@ -0,0 +1,411 @@
...
+    MakeTransferScreen(
+        uiState = MakeTransferState.Error("An error occurred"),
+        showTransactionStatus = ShowTransactionStatus(
+            showSuccessStatus = false,
+            showErrorStatus = false
+        ),
+        makeTransfer = { _, _ -> }
+    )
+}
+
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt:413]: use preview parameter annotation

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt [hunk 7]
```diff
@@ -0,0 +1,411 @@
...
+            MakeTransferBottomSheetContent(
+                showBottomSheet,
+                toClientId,
+                resultName,
+                externalId,
+                transferAmount,
+                showTransactionStatus,
+                makeTransfer = makeTransfer
+            )
```
Comment (AvneetSingh2001) [inline mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt:104]: directly pass these values to the composable. 
also always pass by naming the parameter, link `showBottomSheet = uiState.showBottomSheet` for functions for more good visibility of code.

File: mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt (hunk 4 above)
Comment (Aditya-gupta99) [inline mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt:282]: Use `stringResource`

File: mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt (hunk 3 above)
Comment (Aditya-gupta99) [inline mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt:266]: Use `stringResource`

File: mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt [hunk 8]
```diff
@@ -0,0 +1,411 @@
...
+fun PreviewWithMakeTransferContentError() {
+    MakeTransferScreen(
+        uiState = MakeTransferState.Error("An error occurred"),
+        showTransactionStatus = ShowTransactionStatus(
+            showSuccessStatus = false,
+            showErrorStatus = false
+        ),
+        makeTransfer = { _, _ -> }
+    )
+}
```
Comment (AvneetSingh2001) [inline mifospay/src/main/java/org/mifospay/common/ui/MakeTransferScreen.kt:412]: you can use preview parameters

**Review by AvneetSingh2001 — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifospay/common/presenter/MakeTransferViewModel.kt [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+import org.mifospay.core.data.domain.usecase.client.SearchClient
+import org.mifospay.data.local.LocalRepository
+import javax.inject.Inject
+
+@HiltViewModel
+class MakeTransferViewModel @Inject constructor(
+    private val mUsecaseHandler: UseCaseHandler,
+    private val searchClientUseCase: SearchClient,
+    private val transferFundsUseCase: TransferFunds,
+    private val localRepository: LocalRepository?
```
Comment (Aditya-gupta99) [inline mifospay/src/main/java/org/mifospay/common/presenter/MakeTransferViewModel.kt]: Remove nullable

**Review by Aditya-gupta99 — COMMENTED** [review]
Can you make the require changes

**Pull request merged**